src/test/files/java/RestCallDeprecationCheck.java

```diff
@@ -0,0 +1,10 @@
+import org.mule.api.annotations.Connector;
+import org.mule.api.annotations.Processor;
+
+@Connector
+public abstract class RestCallDeprecationCheck {
+
+    @Processor
+    @RestCall(uri = "http://mulesoft.com/api/aService/?someParam=0")
+    public abstract void failingMethod(); // Noncompliant {{@RestCall should be removed from processor 'failingMethod' as it is deprecated.}}
```

Comment:
`Noncompliant` is a valid Sonar argument to declare issues (failures) that are expected to be raised by the check rule (RestCallDeprecationCheck) as comments. It is mandatory that the message defined there **exactly matches** the one defined in the check class. Otherwise, the test will fail stating that, for example, is expecting "A" but got "A+".

Check this link > https://github.com/QualInsight/qualinsight-libs-sonarqube-test